### PR TITLE
Admin Panel Stops Adding

### PR DESCRIPTION
**Pull request by quuu**
- Created functionality to add stops through the admin panel

- Ability to click on the map and get the coordinates of the clicked location

- POSTS to existing endpoint to store into postgres

- Added `string` field to stop go struct, temporary fix

- Added more fields to the routes admin panel to display more information about the routes

TODO:
- Ability to edit an existing stop
- More error checking
- Document a little more
- Stop enabled or disabled
- Route(s) the stop is associated with

### CODE REVIEW

File: frontend/src/assets/vars.scss [hunk 1]
```diff
@@ -0,0 +1,24 @@
+@import "~bulma/sass/utilities/_all";
+
+$primary: #ED1C24;
+$link: red;
```
[the remark this answers is not in the document]
Comment (quuu) [reply frontend/src/assets/vars.scss]: Fixed

File: frontend/src/assets/vars.scss [hunk 2]
```diff
@@ -0,0 +1,24 @@
+@import "~bulma/sass/utilities/_all";
+
+$primary: #ED1C24;
```
[the remark this answers is not in the document]
Comment (quuu) [reply frontend/src/assets/vars.scss]: Fixed

File: frontend/src/assets/vars.scss [hunk 3]
```diff
@@ -0,0 +1,24 @@
+@import "~bulma/sass/utilities/_all";
```
[the remark this answers is not in the document]
Comment (quuu) [reply frontend/src/assets/vars.scss]: Fixed

File: frontend/src/assets/vars.scss [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+$primary: #ED1C24;
+$link: red;
+$primary-invert: findColorInvert($primary);
+$twitter: #4099FF;
```
[the remark this answers is not in the document]
Comment (quuu) [reply frontend/src/assets/vars.scss]: Fixed

File: frontend/src/assets/vars.scss [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+$twitter-invert: findColorInvert($twitter);
```
[the remark this answers is not in the document]
Comment (quuu) [reply frontend/src/assets/vars.scss]: As discussed with @jlyon1, not an "issue" as it's how Bulma named the function

File: frontend/src/assets/vars.scss [hunk 6]
```diff
@@ -0,0 +1,24 @@
+@import "~bulma/sass/utilities/_all";
+
+$primary: #ED1C24;
+$link: red;
+$primary-invert: findColorInvert($primary);
```
[the remark this answers is not in the document]
Comment (quuu) [reply frontend/src/assets/vars.scss]: As discussed with @jlyon1, not an "issue" as it's how Bulma named the function

**Comment by CoreyBuckley** [comment]
Quick note for your last TODO item: my most recent merged PR adds an array of routes a stop is part of, so feel free to use/improve that. The routes are set in store.ts "setRoutesOnStops"

File: stop.go [hunk 7]
```diff
@@ -8,8 +8,8 @@ import (
 // Stop is a place where vehicles frequently stop.
 type Stop struct {
 	ID        int64     `json:"id"`
-	Latitude  float64   `json:"latitude"`
-	Longitude float64   `json:"longitude"`
+	Latitude  float64   `json:"latitude,string"`
```
Comment (jlyon1) [inline stop.go]: This shouldn't have to be changed on the backend, the types on the frontend should match those on the backend

File: frontend/src/main.ts [hunk 8]
```diff
@@ -18,6 +18,7 @@ import App from '@/App.vue';
 import store from '@/store';
 import router from '@/index';
 
+Vue.use(Buefy);
```
Comment (jlyon1) [inline frontend/src/main.ts]: Duplicate Buefy Import

File: api/api.go [hunk 9]
```diff
@@ -135,6 +135,7 @@ func New(cfg Config, ms shuttletracker.ModelService, msg shuttletracker.MessageS
 
 	r.Get("/", api.IndexHandler)
 	r.Get("/about", api.IndexHandler)
+	r.Get("/test", api.IndexHandler)
```
Comment (jlyon1) [inline api/api.go]: This should be removed

File: api/routes.go [hunk 10]
```diff
@@ -94,7 +97,12 @@ func (api *API) RoutesEditHandler(w http.ResponseWriter, r *http.Request) {
 // StopsCreateHandler adds a new route stop to the database
 func (api *API) StopsCreateHandler(w http.ResponseWriter, r *http.Request) {
 	stop := &shuttletracker.Stop{}
+
 	err := json.NewDecoder(r.Body).Decode(stop)
+	fmt.Println("the stop info is")
+	fmt.Println(*stop)
+	fmt.Println(reflect.TypeOf(*stop))
```
Comment (jlyon1) [inline api/routes.go]: Debug prints should be removed

File: frontend/src/components/admin/stopsEditing.vue [hunk 11]
```diff
@@ -0,0 +1,169 @@
+<template>
+<div class="container">
+    <div class="columns">
+
+        <div class="column is-8">
+            <div v-if="failure" class="notification is-danger">
+                <p>There was an error saving the route.</p>
+            </div>
+            <div v-if="success" class="notification is-success">
+                <p>Route saved successfully</p>
```
Comment (jlyon1) [inline frontend/src/components/admin/stopsEditing.vue]: Should read "Stop saved successfully"

File: frontend/src/components/admin/stopsEditing.vue [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+            <div class="control">
+                <input v-model="stop.description" id="Description" name="Description" type="text" placeholder="Description" class="input " :disabled="!creation">
+            </div>
+            </div>
+
+            <!-- Text input-->
+            <div class="field">
+            <label class="label" for="Latitude">Latitude</label>
+            <div class="control">
+                <input v-model="stop.latitude" id="Latitude" name="Latitude" type="text" placeholder="Latitude" class="input " :disabled="!creation">
```
Comment (jlyon1) [inline frontend/src/components/admin/stopsEditing.vue]: Reactivity doesn't appear to be working on these textboxes

File: frontend/src/components/admin/stopsEditing.vue [hunk 13]
```diff
@@ -0,0 +1,169 @@
...
+            </div>
+            </div>
+
+            <!-- Text input-->
+            <div class="field">
+            <label class="label" for="Longitude">Longitude</label>
+            <div class="control">
+                <input v-model="stop.longitude" id="Longitude" name="Longitude" type="text" placeholder="Longitude" class="input " :disabled="!creation">
```
Comment (jlyon1) [inline frontend/src/components/admin/stopsEditing.vue]: Same with this one

File: frontend/src/components/admin/stopsEditing.vue (hunk 12 above)
Comment (quuu) [reply frontend/src/components/admin/stopsEditing.vue]: Reactive form was not implemented, don't think it's an issue?

File: frontend/src/components/admin/stopsEditing.vue (hunk 13 above)
Comment (quuu) [reply frontend/src/components/admin/stopsEditing.vue]: Reactive form was not implemented, don't think it's an issue?

File: frontend/src/components/admin/stopsEditing.vue (hunk 11 above)
Comment (quuu) [reply frontend/src/components/admin/stopsEditing.vue]: Resolved in most recent commit

File: api/routes.go (hunk 10 above)
Comment (quuu) [reply api/routes.go]: Resolved in most recent commit

File: api/api.go (hunk 9 above)
Comment (quuu) [reply api/api.go]: Resolved in most recent commit

Was going to be for HTML schedules endpoint, but will remove for now

File: frontend/src/components/admin/stopsEditing.vue (hunk 12 above)
Comment (jlyon1) [reply frontend/src/components/admin/stopsEditing.vue]: well the v-model is supposed to make the box reactive, It is confusing to have -1 in the box when that is not the value being posted.

File: frontend/src/main.ts (hunk 8 above)
Comment (quuu) [reply frontend/src/main.ts]: Resolved

File: frontend/src/components/admin/stopsEditing.vue (hunk 12 above)
Comment (quuu) [reply frontend/src/components/admin/stopsEditing.vue]: Resolved

File: frontend/src/components/admin/stopsEditing.vue (hunk 13 above)
Comment (quuu) [reply frontend/src/components/admin/stopsEditing.vue]: Resolved

**Pull request merged**